Orange/widgets/visualize/owmosaic.py

```diff
@@ -447,13 +445,19 @@ def updateGraph(self, data=-1, subsetData=-1, attrList=-1, **args):
         if data == -1:
             data = self.data
 
+        if data != None:
+            if any(attr.is_continuous for attr in data.domain):
+                self.discrete_data = Discretize(method=EqualFreq(n=4))(data)
+            else:
+                self.discrete_data = self.data
+
         if subsetData == -1:
             subsetData = self.subset_data
 
         if attrList == -1:
             attrList = [self.variable1, self.variable2, self.variable3, self.variable4]
 
-        if data == None: return
+        if self.discrete_data == None: return
```

Comment:
Use `is` instead of `==`. This has already been "wrong" before, but you can fix it while you're touching this line.

(Also, avoid compound statements in a single line. And, again, this return has already been in the same line before you modified the code, so it's not your fault. :))